chain/epoch_manager/src/tests/mod.rs

```diff
@@ -2391,3 +2392,75 @@ fn test_epoch_validators_cache() {
         epoch_manager.epoch_validators_ordered_unique.get(&epoch_id).unwrap();
     assert_eq!(*epoch_validators_unique, *epoch_validators_unique_in_cache);
 }
+
+#[test]
+fn test_chunk_producers() {
+    let amount_staked = 1_000_000;
+    // Make sure that last validator has at least 160/1'000'000  / num_shards of stake.
+    // We're running with 2 shards and test1 + test2 has 2'000'000 tokens - so chunk_only should have over 160.
+    let validators = vec![
+        ("test1".parse().unwrap(), amount_staked),
+        ("test2".parse().unwrap(), amount_staked),
+        ("chunk_only".parse().unwrap(), 200),
+        ("not_enough_producer".parse().unwrap(), 100),
+    ];
+
+    #[cfg(not(feature = "protocol_feature_chunk_only_producers"))]
+    let protocol_version = PROTOCOL_VERSION;
+    #[cfg(feature = "protocol_feature_chunk_only_producers")]
+    let protocol_version = std::cmp::max(
+        PROTOCOL_VERSION,
+        ProtocolFeature::protocol_version(ProtocolFeature::ChunkOnlyProducers),
+    );
```

Comment:
I don't understand why we need this. Isn't this always equivalent to just this?
```
let protocol_version = PROTOCOL_VERSION;
```
If it is equivalent, I think you can also get rid of `setup_epoch_manager_with_simple_nightshade_config_and_protocol_version`.

Or are you planning to compile with `protocol_feature_chunk_only_producers` feature but without `nightly` feature?

Reply:
yes, I was thinking about testing only protocol_feature_chunk_only_producers -- but thats's probably an overkill..

I'll simplify as you suggest.